### PR TITLE
Added request object construction to doc gen

### PR DESCRIPTION
**Pull request by michaelbausor**
Construct request object in documentation
Updated tests

### CODE REVIEW

File: vgen/src/main/java/io/gapi/vgen/java/JavaContextCommon.java [hunk 1]
```diff
@@ -212,7 +221,51 @@ public static Variable s_newVariable(
 
       public abstract Builder setParams(ImmutableList<Variable> params);
 
-      public Builder setParams(JavaGapicContext languageProvider, ImmutableList<Field> fields) {
+      public Builder setParams(JavaGapicContext languageProvider, Iterable<Field> fields) {
```
Comment (garrettjonesgoogle) [inline vgen/src/main/java/io/gapi/vgen/java/JavaContextCommon.java]: `languageProvider` is no longer the correct name for the parameter - it should be `context`

File: vgen/src/main/resources/io/gapi/vgen/java/main.snip [hunk 2]
```diff
@@ -626,14 +636,58 @@
...
 @private initValue(param, ApiName)
-  @if param.hasFormattingConfig() {
+  @if param.hasFormattingConfig()
     {@ApiName}.{@formatResourceFunctionName(param.getFormattingConfig())}({@formatResourceFunctionArgs(param.getFormattingConfig())})
   @else
     {@context.zeroValue(param.getType)}
   @end
 @end
 
+@snippet setCallNameFromParam(param)
```
Comment (garrettjonesgoogle) [inline vgen/src/main/resources/io/gapi/vgen/java/main.snip]: I would prefer not to repeat the structure of setCallName in common.snip.

**Comment by michaelbausor** [comment]
PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**